pootle/apps/pootle_format/utils.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import os
+from collections import OrderedDict
+
+from .exceptions import UnrecognizedFiletype
+
+
+class ProjectFiletypes(object):
+
+    def __init__(self, project):
+        self.project = project
+
+    @property
+    def filetypes(self):
+        return self.project.filetypes.all()
+
+    def choose_filetype(self, filename):
+        ext = os.path.splitext(filename)[1][1:]
+        formats = self.project.filetypes.filter(
+            extension__name=ext)
+        if formats.count():
```

Comment:
yep - ill change - i never rem which is the cheapest `models` `models.exists` or `models.count` - not the last tho i think